### DISCUSSION TITLE
Define a basis refractive index for the whole board different from 1

### DISCUSSION THREAD

**Discussion by mcornillon** [ricktu288/ray-optics#146] [Ideas]
Hi ! Big thanks for your project, that's really amazing
I'm using it to make schematics for underwater photography.
And I'd be really interested to have the possibility to set the "basis refractive index" to a value different from 1 (1.33 in my case), instead of having to define a half-plane water-air, which comes with more refractions & reflections...

Any possibility already ?
Or a possible evolution ?

Thanks !

**Comment by ricktu288** [comment]
Note that all the "refractive index" options in this simulator are relative to the value outside the material (see #51, #137), so even if you fill the space with a material (like water), you still need to set the refractive index of the material immersed in the water to relative to the water. For example, if you want to simulate an air bubble, you need to set the bubble to have index 1/1.33. So actually whether you filled the entire space with water will not change any simulation result. The only thing that will change is the color representation of material as shown in the simulator.

For now, you may put two half-plane glasses together to achieve filling the entire space.

**Reply by mcornillon** [reply]
Yes, ok, thanks for the explanation.
And thanks for the idea of two half-planes, simple, and allows me to tackle (by deactivating one half) to simulate "split-shots over/under" use cases
